build/steps-build-android.yml

```diff
@@ -0,0 +1,117 @@
+parameters:
+- name: androidKeyStoreFile
+  type: string
+  default: ''
+- name: pathToSrc
+  type: string
+  default: ''
+- name: solutionName
+  type: string
+  default: ''
+- name: pathToAndroidManifest
+  type: string
+  default: ''
+
+steps:
+- template: templates/dotnet-install.yml
+
+#-if false
+- download: current
+  condition: eq('GeneratedApp', '${{ parameters.solutionName }}')
+  artifact: GeneratedApp
+#-endif
+
+- task: gitversion/setup@0
+  retryCountOnTaskFailure: 3
+  inputs:
+    versionSpec: '5.10.1'
+  displayName: 'Install GitVersion'
+
+- task: gitversion/execute@0
+  inputs:
+    useConfigFile: true
+    configFilePath: $(Build.SourcesDirectory)/gitversion.yml
+  displayName: 'Calculate Version'
+
+- task: DownloadSecureFile@1
+  name: keyStore
+  displayName: "Download Keystore from Secure Files"
+  inputs:
+    secureFile: ${{ parameters.androidKeyStoreFile }}
+
+- task: android-manifest-package-name@1
+  displayName: "Update Bundle Identifier"
+  inputs:
+    sourcePath: ${{ parameters.pathToAndroidManifest }}
+    packageName: $(ApplicationIdentifier)
+    printFile: true
+
+- task: android-manifest-version@1
+  displayName: "Bump Android Manifest Version"
+  inputs:
+    sourcePath: ${{ parameters.pathToAndroidManifest }}
+    versionCode: $(PreReleaseNumber)
+    versionName: $(MajorMinorPatch)
+    printFile: true
+
+- script: >
```

Comment:
Why do we have two scripts by the way ?

Reply:
Nvm its for light build.

Reply:
Still I don't get why you need to split ? Can't we make something to refacto this in a single script ?

Reply:
Ok as discussed we can optimize that by dynamically set a variable for publish vs build.

Reply:
Done! :)